src/NLog/Targets/DatabaseParameterInfo.cs

```diff
@@ -98,6 +98,20 @@ public DatabaseParameterInfo(string parameterName, Layout parameterLayout)
         /// <docgen category='Parameter Options' order='10' />
         [DefaultValue(0)]
         public byte Scale { get; set; }
+
+        /// <summary>
+        /// Gets or sets the column type of the database.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the database column type property.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbTypePropertyName { get; set; }
```

Comment:
I get your point. But it's still more generic as I can set the parameter type for every db vendor from the config, and it's not limited to clobs. Also, I think most of the db vendors uses the 
IDbDataParameter and not a special property.  

The propery name from the config is a bit difficult too understand, so I prefer for that a special case. 

Scanning the properties sounds a bit magic and performance problematic.  Do you agree?